### ISSUE TITLE
Why does typing `git add *` not add .gitignore file into the staging area

### ISSUE THREAD

**Issue by obrona** [nus-cs2103-AY2425S1/forum#228]
Why does typing `git add *` not add `.gitignore` file into the staging area ?

**Comment by suhayl13** [comment]
Hi! I searched up the web and found possible reasons why this could happen. 

If we type `git add *`, Git will add all files in the current directory to the staging area except for those that are listed in the `.gitignore` file or those that match the patterns defined in it. 

The .gitignore file may not be added to the staging area when you run `git add *` because of the following reasons:
- The `.gitignore`file is ignored by another `.gitignore` entry
- The `.gitignore` file is not in the directory where you're running `git add *`
- The `.gitignore` file was already added and committed previously

To explicitly add `.gitignore` file, we can run `git add .gitignore`. To add all modified files including `.gitignore` we can simply run `git add .`

Hope this helps!

**Comment by yadobler** [comment]
**TL;DR** because `*` by default does not include _hidden files_ (starting with `.`). to include them you need to also run `git add .*`.
---

`*` wildcard (globbing) is not a git-specific thing, but dependent on the shell you are running. Shells like _bash_ and _zsh_ define the behaviour of what `*` globs. You can edit them in their respective config / `rc` files.

The default behaviour of `*` only includes non-hidden files, which means anything starting with `.`, ie `.gitignore`, will be ignored.

---

- `git add -A` or `git add -All` If you're trying to add *ALL* files (that are not excluded by `.gitignore`)
- `git add -u` if you only want to add all **currently tracked** files (**This is most often used**)
- `git add -f <File>` to forcefully stage a _currently ignored_ file.

**Comment by obrona** [comment]
Thanks guys I get it know
